File: api/src/main/java/com/github/av2/api/model/Order.java
package com.github.av2.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Order information")
public class Order {
    @Schema(description = "The unique identifier for the order", required = true)
    private Integer orderId;
    
    @Schema(description = "The ID of the branch that placed the order", required = true)
    private Integer branchId;
    
    @Schema(description = "The date and time when the order was placed", required = true)
    private String orderDate;
    
    @Schema(description = "The current status of the order", required = true)
    private String status;
    
    @Schema(description = "Additional notes about the order")
    private String notes;
}

File: api/Models/Order.cs
using System;

namespace OctoCat.Api.Models
{
    public class Order : IEntity<int>
    {
        public int OrderId { get; set; }
        public int BranchId { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public DateTime OrderDate { get; set; }
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;

        // Implement IEntity interface
        public int Id 
        { 
            get => OrderId;
            set => OrderId = value;
        }
    }
}